pyrate/core/timeseries.py

```diff
@@ -302,6 +303,102 @@ def _solve_ts_lap(nvelpar, velflag, ifgv, mat_b, smorder, smfactor, sel, vcmt):
     return tsvel
 
 
+def linear_rate_pixel(y, t):
+    """
+    Calculate best fitting linear rate to cumulative displacement
+    time series for one pixel using linear regression.
+
+    :param ndarray y: 1-dimensional vector of cumulative displacement time series
+    :param ndarray t: 1-dimensional vector of cumulative time at each epoch
+
+    :return: linrate: Linear rate; gradient of fitted line.
+    :rtype: float
+    :return intercept: Y-axis intercept of fitted line at t = 0
+    :rtype: float
+    :return: rsquared: R-squared value of the linear regression
+    :rtype: float
+    :return: error: Standard error of the linear regression
+    :rtype: float
+    :return: samples: Number of observations used in linear regression
+    :rtype: int
+    """
+
+    # test for equal length of input vectors
+    if len(y) != len(t):
+        raise TimeSeriesError("linear_rate_pixel: y and t are not equal length")
+
```

Comment:
The point of the check is to do error checking on the input in case someone calls this function directly, outside of the context of the workflow. The raised exception gives a more contextual message than the `linregress` exception. How much time do we expect to save by removing this check on say, 1 Million function calls (i.e. a spatial array of 1000 x 1000)?

Reply:
Good discussion. Another point of view could be to use exception handling to catch and reraise the linregress exception and turn it into something more meaningful to the caller - and at the same time avoiding doing the same check 1 million times. Admittedly, whether this is a performance issue should be substantiated by profiling.

Reply:
Thanks for commit 72aae9e @basaks , which I think is a good compromise. I have also used @uniomni's suggestion to re-raise the exception in the `linear_rate_pixel` function to give a more context specific exception message. Good outcome!